.github/workflows/todo.yml

```diff
@@ -2,6 +2,19 @@ name: To Do app CI & CD
 on: [push, pull_request]
 
 jobs:
+
+  deployingtoheroku:
+    name: heroku
+    runs-on: ubuntu-latest
```

Comment:
The way that you've set this up, you will update your app at https://todo-app-london.herokuapp.com every time anyone pushes to any branch, whether or not the tests pass.

That's not what we want - we don't want to push a broken version and some branches will have experimental work or stuff in progress. The exercise tells you to configure this to run only for pushes to the main branch and only once the tests have passed.